### ISSUE TITLE
Resource language fallback should be the default content language (if possible)

### ISSUE THREAD

**Issue by bep** [gohugoio/hugo#11840]
```handlebars
func TestBundleResourceLanguageBestMatch(t *testing.T) {
	files := `
-- hugo.toml --
defaultContentLanguage = "fr"
defaultContentLanguageInSubdir = true
[languages]
[languages.en]
weight = 1
[languages.fr]
weight = 2
[languages.de]
weight = 3
-- layouts/index.html --
{{ $bundle := site.GetPage "bundle" }}
{{ $r := $bundle.Resources.GetMatch "*.txt" }}
{{ .Language.Lang }}: {{ $r.RelPermalink }}|{{ $r.Content }}
-- content/bundle/index.fr.md --
---
title: "Bundle Fr"
---
-- content/bundle/index.en.md --
---
title: "Bundle En"
---
-- content/bundle/index.de.md --
---
title: "Bundle De"
---
-- content/bundle/data.fr.txt --
Data fr
-- content/bundle/data.en.txt --
Data en

`
	b := Test(t, files)

	b.AssertFileContent("public/fr/index.html", "fr: /fr/bundle/data.fr.txt|Data fr")
	b.AssertFileContent("public/en/index.html", "en: /en/bundle/data.en.txt|Data en")
	b.AssertFileContent("public/de/index.html", "de: /de/bundle/data.en.txt|Data en")
}

```

@jmooring if I could borrow your eyes for a minute.

The above test passes in the master branch, which surprises me a little. I guess most people have their `defaultContentLanguage` defined as their first language.

The surprise is this:

```bash
b.AssertFileContent("public/de/index.html", "de: /de/bundle/data.en.txt|Data en")
```

What fallback text file would you expect in the de bundle?

**Comment by jmooring** [comment]
Yes, that is surprising. I would expect to see "Data fr".

We should use the same fallback logic for resource selection that we use for `lang.Translate`. 

In the example above, `lang.Translate` for a word missing from `i18n/de.toml` falls back to the value in `i18n/fr.toml` (the default content language). This is the correct behavior; the language weight is irrelevant.

I may be forgetting some details, but from a site/theme author's perspective the language weight is only useful for things like ordering `.Page.Translations` and `.Page.AllTranslations` for language switchers, etc.

**Comment by bep** [comment]
>I may be forgetting some details, but from a site/theme author's perspective the language weight is only useful for things like ordering .Page.Translations and .Page.AllTranslations for language switchers, etc.

To sum up my understanding:

* Translations/AllTranslations are ordered by the language order (weight etc.)
* All other language ordering (including the bug above) is ordered first by the default content language, then by weight etc.

**Comment by jmooring** [comment]
Remember <https://github.com/gohugoio/hugo/issues/9216>? My fix for that [#10909](https://github.com/gohugoio/hugo/pull/10909) was limited to translation.

The behavior described above is the same problem. We are falling back to "en", not the default content language, not the "lightest" weighted language. This is evident if you change the test case to use "es" files instead of "en" files.
